keras/applications/regnet.py

```diff
@@ -0,0 +1,1349 @@
+  Y block and 6.4 giga flops (64 hundred million flops). 
+
+  Args:
+    include_top: Whether to include the fully-connected
+        layer at the top of the network. Defaults to True.
+    weights: One of `None` (random initialization),
+          "imagenet" (pre-training on ImageNet),
+          or the path to the weights file to be loaded. Defaults to "imagenet".
+    input_tensor: Optional Keras tensor
+        (i.e. output of `layers.Input()`)
+        to use as image input for the model.
+    input_shape: Optional shape tuple, only to be specified
+        if `include_top` is False.
+        It should have exactly 3 inputs channels.
+    pooling: Optional pooling mode for feature extraction
+        when `include_top` is `False`. Defaults to None.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+    classes: Optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified. Defaults to 1000 (number of
+        ImageNet classes).
+    classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        Defaults to "softmax".
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+  Returns:
+    A `keras.Model` instance.
+"""
+
+def PreStem():
+  """Rescales and normalizes inputs to [0,1] and ImageNet mean and std.
+  
+  Args:
+    x: input tensor
+    
+  Returns:
+    Rescaled and normalized tensor
+  """
+
+  def apply(x):
+    IMGNET_MEAN = tf.constant([0.485, 0.456, 0.406])
+
+    IMGNET_STD = tf.constant([0.229, 0.224, 0.225])
+
+    x = layers.Rescaling(scale=1./255., name="prestem_rescaling")(x)
+    x = layers.Normalization(mean=IMGNET_MEAN, 
+          variance=tf.math.square(IMGNET_STD), name="prestem_normalization")(x)
+
+    return x
+  return apply
+
+def Stem():
+  """Implementation of RegNet stem. (Common to all model variants)
+  
+  Args:
+    x: Input tensor. Should be 224x224, rescaled and normalized to [0,1].   
+
+  Returns:
+    Output tensor of the Stem
+  """
+  def apply(x):
+    x = layers.Conv2D(32, (3, 3), strides=2, use_bias=False, name="stem_conv")(x)
+    x = layers.BatchNormalization(momentum=0.9, epsilon=1e-5, name="stem_bn")(x)
+    x = layers.ReLU(name="stem_relu")(x)
+
+    return x
+  return apply
+
+
+def XBlock(filters_in,
+           filters_out,
+           group_width,
+           stride=1,
+           name=""):
```

Comment:
The default name can't really be `""`. Use `name=None` in the signature, then if `None` convert it to an autogenerated string via `backend.get_uid("xblock")`

Reply:
I'll change the default to None. However, in the code all names are in a prefixed manner and using `get_uid` won't be necessary. Example:

```python

def XBlock(name=None):
  ...
  x = conv_3x3(x)
  x = layers.BatchNormalization(
        momentum=0.9, epsilon=1e-5, name=name + "_conv_3x3_bn")(x)
  x = layers.ReLU(name=name + "_conv_3x3_relu")(x)
  ...

def Stage(name=None):
  ...
  for i in range(depth - 1):
        x = XBlock(filters_out, filters_out,
                   group_width, name=name + "_XBlock_" + str(i))(x)
  ...


def RegNet(model_name=None):
  ...
  for num_stage in range(4):
    x = Stage(...,  name=model_name + "_Stage_" + str(num_stage))(x)
    
```

Please let me know if the above isn't the best way. I'll change it asap.

Reply:
You still need to set the prefix default with get_uid.

```
if name is None:
  name = backend.get_uid("XBlock")
```